### ISSUE TITLE
In-Stock Items Labeled as Out of Stock. 

### ISSUE THREAD

**Issue by co50** [woocommerce/woocommerce#19647]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [X] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [X] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. The issue appears to have no known formula to reproduce. Its randomness is part of the bug.
2. Import whole catalog via CSV Import Suite (bug not related to CSV Import Suite)
3. Random in-stock products show "OUT OF STOCK" when browsing archives, and in admin products list.
4. Single product page shows correct (in stock) inventory. Admin details page shows correct stock info.
5. Merely clicking UPDATE in admin on product's page w/o changing anything corrects the incorrect OUT OF STOCK messaging in both areas.

See video made of this issue:  https://youtu.be/74h_MfTatxk

## Expected/actual behavior

When I follow those steps, I see...

Items that are in-stock being marked as OUT OF STOCK

I was expecting to see...

Stock labeling that correctly matches the items' inventory.

- [X] This bug happens with only WooCommerce plugin active
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [X] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: https://frankmorrow.com
Site URL: https://frankmorrow.com
WC Version: 3.3.4
Log Directory Writable: ✔
WP Version: 4.9.5
WP Multisite: –
WP Memory Limit: 384 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.2.2
PHP Post Max Size: 32 MB
PHP Time Limit: 0
PHP Max Input Vars: 1000
cURL Version: 7.19.7
NSS/3.27.1

SUHOSIN Installed: –
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.4
WC Database Prefix: wordpress_
MaxMind GeoIP Database: ✔
Total Database Size: 866.01MB
Database Data Size: 826.12MB
Database Index Size: 39.89MB
wordpress_woocommerce_sessions: Data: 32.14MB + Index: 0.30MB
wordpress_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wordpress_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.01MB
wordpress_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB
wordpress_woocommerce_order_items: Data: 0.35MB + Index: 0.16MB
wordpress_woocommerce_order_itemmeta: Data: 2.18MB + Index: 1.58MB
wordpress_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB
wordpress_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB
wordpress_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wordpress_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB
wordpress_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wordpress_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wordpress_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wordpress_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wordpress_commentmeta: Data: 0.08MB + Index: 0.07MB
wordpress_comments: Data: 1.45MB + Index: 0.56MB
wordpress_gml_locations: Data: 0.00MB + Index: 0.00MB
wordpress_gml_settings: Data: 0.00MB + Index: 0.00MB
wordpress_imagify_files: Data: 0.02MB + Index: 0.08MB
wordpress_imagify_folders: Data: 0.02MB + Index: 0.03MB
wordpress_layerslider: Data: 0.02MB + Index: 0.00MB
wordpress_links: Data: 0.00MB + Index: 0.00MB
wordpress_options: Data: 663.60MB + Index: 1.28MB
wordpress_postmeta: Data: 108.74MB + Index: 28.68MB
wordpress_posts: Data: 11.81MB + Index: 3.92MB
wordpress_rg_form: Data: 0.00MB + Index: 0.00MB
wordpress_rg_form_meta: Data: 0.04MB + Index: 0.00MB
wordpress_rg_form_view: Data: 0.38MB + Index: 0.23MB
wordpress_rg_incomplete_submissions: Data: 0.02MB + Index: 0.02MB
wordpress_rg_lead: Data: 0.10MB + Index: 0.02MB
wordpress_rg_lead_detail: Data: 0.27MB + Index: 0.31MB
wordpress_rg_lead_detail_long: Data: 0.07MB + Index: 0.00MB
wordpress_rg_lead_meta: Data: 0.00MB + Index: 0.00MB
wordpress_rg_lead_notes: Data: 0.00MB + Index: 0.00MB
wordpress_srb_blacklist: Data: 0.02MB + Index: 0.03MB
wordpress_taxonomymeta: Data: 0.00MB + Index: 0.01MB
wordpress_termmeta: Data: 0.13MB + Index: 0.13MB
wordpress_terms: Data: 0.01MB + Index: 0.04MB
wordpress_term_relationships: Data: 0.80MB + Index: 1.48MB
wordpress_term_taxonomy: Data: 0.01MB + Index: 0.02MB
wordpress_translations: Data: 0.52MB + Index: 0.12MB
wordpress_usermeta: Data: 1.22MB + Index: 0.52MB
wordpress_users: Data: 0.05MB + Index: 0.05MB
wordpress_wc_download_log: Data: 0.02MB + Index: 0.03MB
wordpress_wc_webhooks: Data: 0.02MB + Index: 0.02MB
wordpress_woos_search_analytics: Data: 1.52MB + Index: 0.00MB
wordpress_woos_search_trends: Data: 0.02MB + Index: 0.02MB
wordpress_wpsl_stores: Data: 0.00MB + Index: 0.00MB
wordpress_yoast_seo_links: Data: 0.02MB + Index: 0.02MB
wordpress_yoast_seo_meta: Data: 0.33MB + Index: 0.00MB

### Post Type Counts ###

attachment: 26842
avada_portfolio: 168
etheme_portfolio: 171
log-update: 9
nav_menu_item: 87
option-tree: 2
page: 21
plugins-inventory: 46
post: 3
postman_sent_mail: 250
product: 5356
product_variation: 11659
shop_order: 1259
shop_order_refund: 46
slide: 1
staticblocks: 1
testimonial: 2
vtmin-rule: 1
wpsl_stores: 14

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (40) ###

ManageWP - Worker: by ManageWP – 4.5.0
Gravity Forms: by rocketgenius – 2.2.6
Admin Menu Editor Pro: by Janis Elsts – 2.6.3
Akismet Anti-Spam: by Automattic – 4.0.3
Fusion Builder: by ThemeFusion – 1.4.2
Fusion Core: by ThemeFusion – 3.4.2
IgniteWoo Updater: by IgniteWoo.com – 2.1.11
Image Sizes: by Nazmul Ahsan – 1.2.1
Imagify: by WP Media – 1.7
Invisible reCaptcha: by Mihai Chelaru – 1.2.1
Mobile Address Bar Changer: by Anshul Labs – 3.0
My Custom Functions: by Arthur Gareginyan – 4.13
PDF Product Catalog 2 for WooCommerce: by Kyriakos Ktorides – 2.3.3 – Not tested with the active version of WooCommerce
Post SMTP: by Jason Hendriks
Yehuda Hassine – 1.8.3

WooCommerce Product Filter: by Mihajlovic Nenad – 6.5.6 – Not tested with the active version of WooCommerce
Really Simple SSL: by Rogier Lankhorst – 2.5.26
Hide SEO Bloat: by SO WP – 3.5.0
UberMenu 3 - The Ultimate WordPress Mega Menu: by Chris Mavricos
SevenSpark – 3.3.1

Ultimate Branding: by WPMU DEV – 1.9.3
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 2.14.3.22

WooCommerce Variation Swatches: by ThemeAlien – 1.0.3 – Not tested with the active version of WooCommerce
WC Password Strength Settings: by Daniel Santoro – 2.0.2
Woo Custom Fee: by AcvSoft – 1.6
Woo Search Box: by Guaven Labs – 1.9.0 – Not tested with the active version of WooCommerce
WooCommerce Bulk Pricing: by WP Lab – 1.9.4
WooCommerce Authorize.Net CIM Gateway: by SkyVerge – 2.9.0 – 2.10.0 is available
WooCommerce Google Analytics Pro: by SkyVerge – 1.5.0
WooCommerce Product CSV Import Suite: by WooCommerce – 1.10.17
WooCommerce Shipment Tracking: by WooCommerce – 1.6.10
WooCommerce USPS Shipping: by WooCommerce – 4.4.15
WooCommerce UPS Shipping Pro: by IgniteWoo.com - Custom Add-ons for WooCommerce! – 3.5 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.3.4
Yoast SEO: by Team Yoast – 7.2
WordPress Updates Log: by Sofiane.A – 1.0.1
WP Nag Hide: by Milton Keynes Web Design – 1.0
WP Rocket: by WP Media – 3.0.2
WP Store Locator: by Tijmen Smit – 2.2.14
Yoast SEO: WooCommerce: by Team Yoast – 6.3 – Not tested with the active version of WooCommerce
YITH WooCommerce Best Sellers Premium: by YITHEMES – 1.1.3
YITH WooCommerce Customize My Account Page: by YITHEMES – 2.2.8

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #650 - /
Cart: #651 - /cart/
Checkout: #652 - /checkout/
My account: #653 - /my-account/
Terms and conditions: #710 - /policies-terms/

### Theme ###

Name: Avada
Version: 5.4.2
Author URL: http://themeforest.net/user/ThemeFusion
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: Avada/woocommerce/cart/cart.php
Avada/woocommerce/checkout/form-pay.php
Avada/woocommerce/checkout/review-order.php
Avada/woocommerce/checkout/thankyou.php
Avada/woocommerce/loop/loop-start.php
Avada/woocommerce/single-product/add-to-cart/variable.php
Avada/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
Avada/woocommerce/single-product/short-description.php
Avada/woocommerce/single-product/tabs/additional-information.php
Avada/woocommerce/single-product/tabs/description.php


### Authorize.Net CIM ###

Environment: Production
Tokenization Enabled: ✔
Debug Mode: Off
`
```
</details>

**Comment by kloon** [comment]
@co50 I see you mention importing via the CSV Import Suite which is not part of core. What happens when you make use of the built in core CSV importer for products? Also would you mind providing a sample of the file you are importing so we can test this and inspect the file? Chances are this is an issue with the CSV Import Suite hence why I am asking you to test this with the built in importer instead.

**Comment by co50** [comment]
I can't really do it that because the built in importer is not an exact clone of CSV Import Suite and works differently, supporting fewer columns and advanced import types. We already tried migrating to it in the past and found incompatibilities. Has that changed? Even if the issue can somehow be linked to CSV Import Suite - the data after import is all there, WooCommerce is simply ignoring it or not parsing it correctly. But I'd be happy to provide a copy of the file privately if possible.

**Comment by kloon** [comment]
If the data is there then the CSV Importer is not setting the correct data flag in the code which is why WooCommerce is not picking it up. I suggest you open a ticket with support and have them look into this issue for you as we only cover WooCommerce core related issues on this platform which is only the core CSV Importer and the CSV Import Suite.

**Comment by co50** [comment]
Already have in the past. Was determined to be a core issue. Mike Jolley acknowledged it, and explained why it happens. Many users chimed it that they experience the same thing. Bug never fixed. I think the blame shifting is why this bug has never been resolved in the 2 years since users have been bringing it up. This would be the first time I've ever presented a bug on video to devs of any software I've ever used and it was ignored. Crazy.

**Comment by kloon** [comment]
Could you perhaps provide me with a link to the issue and/or ticket number where you dealt with support?

**Comment by co50** [comment]
Definitely not. I have 60+ clients, all of whom have their own account for WooCommerce plugins. Probably 20+ use CSV Import Suite. I have no clue which client's account was used for the support ticket when it was submitted last year.

We've implemented a workaround that basically forces the UPDATE action whenever someone visits any "out of stock" product. If you guys want to fix the bug, great. If not, cool. But it's a real bug, as I demonstrated clearly on video.
